src/public/services/CreateFormService.ts

```diff
@@ -0,0 +1,50 @@
+import axios from 'axios'
+
+import { DuplicateFormBody } from 'src/app/modules/form/admin-form/admin-form.types'
+import { FormMetaView, IForm, IFormSchema } from 'src/types'
+
+// endpoints exported for testing
+export const ADMIN_FORM_ENDPOINT = '/api/v3/admin/forms'
+
+/**
+ * Duplicates the form
+ * @param formId formId of form to be duplicated
+ * @param duplicateFormBody Title, response mode and relevant information for new form
+ * @returns Metadata of duplicated form for dashboard view
+ */
+export const duplicateForm = async (
+  formId: string,
+  duplicateFormBody: DuplicateFormBody,
+): Promise<FormMetaView> => {
+  return axios
+    .post<FormMetaView>(
+      `${ADMIN_FORM_ENDPOINT}/${formId}/duplicate`,
+      duplicateFormBody,
+    )
+    .then(({ data }) => data)
+}
+
+/**
+ * Creates a new form. This function is called without formId, so the endpoint is just /adminform.
+ * @param newForm Form fields to newly created form
+ * @returns Newly created form.
+ */
+export const createForm = async (
+  newForm: Omit<IForm, 'admin'>,
+): Promise<IFormSchema> => {
+  return axios
+    .post<IFormSchema>(`${ADMIN_FORM_ENDPOINT}`, { form: newForm })
+    .then(({ data }) => data)
+}
+
+/**
+ * Deletes the form with the corresponding formId
+ * @param formId formId of form to delete
+ */
+export const deleteForm = async (
```

Comment:
I think this belongs in `UpdateFormService` actually, wdyt? I would think `CreateFormService` relate to actions that can create forms which don't yet exist, whereas deleting is an action to update the state of an existing form

Reply:
That makes sense! Initially I wanted the service to be in charge of handling the existence of a form i.e. creating and deleting forms but I couldn't think of a good name for it. I'll shift it into `UpdateFormService` then.